simulation/src/test/java/com/palantir/dialogue/core/Strategy.java

```diff
@@ -29,52 +29,49 @@
-        return withDefaults(sim, channelSupplier, configBuilder -> configBuilder
-                .nodeSelectionStrategy(NodeSelectionStrategy.ROUND_ROBIN)
-                .failedUrlCooldown(Duration.ofMillis(200)));
-    }
-
-    private static Channel pinUntilError(Simulation sim, Supplier<Map<String, SimulationServer>> channelSupplier) {
-        return withDefaults(
-                sim,
-                channelSupplier,
-                configBuilder -> configBuilder.nodeSelectionStrategy(NodeSelectionStrategy.PIN_UNTIL_ERROR));
-    }
-
-    private static Channel unlimitedRoundRobin(
-            Simulation sim, Supplier<Map<String, SimulationServer>> channelSupplier) {
-        return withDefaults(sim, channelSupplier, configBuilder -> configBuilder
-                .nodeSelectionStrategy(NodeSelectionStrategy.ROUND_ROBIN)
-                .failedUrlCooldown(Duration.ofMillis(200))
-                .clientQoS(ClientConfiguration.ClientQoS.DANGEROUS_DISABLE_SYMPATHETIC_CLIENT_QOS));
-    }
+public enum Strategy implements GetChannel {
+    CONCURRENCY_LIMITER_ROUND_ROBIN {
+        @Override
+        public Channel getChannel(Simulation sim, Supplier<Map<String, SimulationServer>> servers, int clientNum) {
+            return withDefaults(sim, servers, clientNum, configBuilder -> configBuilder
+                    .nodeSelectionStrategy(NodeSelectionStrategy.ROUND_ROBIN)
+                    .failedUrlCooldown(Duration.ofMillis(200)));
+        }
+    },
+    CONCURRENCY_LIMITER_PIN_UNTIL_ERROR() {
+        @Override
+        public Channel getChannel(
+                Simulation simulation, Supplier<Map<String, SimulationServer>> servers, int clientNum) {
+            return withDefaults(
+                    simulation,
+                    servers,
+                    clientNum,
+                    configBuilder -> configBuilder.nodeSelectionStrategy(NodeSelectionStrategy.PIN_UNTIL_ERROR));
+        }
+    },
+    UNLIMITED_ROUND_ROBIN() {
+        @Override
+        public Channel getChannel(
+                Simulation simulation, Supplier<Map<String, SimulationServer>> servers, int clientNum) {
+            return withDefaults(simulation, servers, clientNum, configBuilder -> configBuilder
+                    .nodeSelectionStrategy(NodeSelectionStrategy.ROUND_ROBIN)
+                    .failedUrlCooldown(Duration.ofMillis(200))
+                    .clientQoS(ClientConfiguration.ClientQoS.DANGEROUS_DISABLE_SYMPATHETIC_CLIENT_QOS));
+        }
+    };
 
     private static Channel withDefaults(
             Simulation sim,
             Supplier<Map<String, SimulationServer>> channelSupplier,
+            int clientNumber,
             UnaryOperator<ClientConfiguration.Builder> applyConfig) {
-        Random pseudo = new Random(3218974678L);
+        // ensures each client doesn't behave identically
+        Random pseudo = new Random(3218974678L + clientNumber);
```

Comment:
What if we add a `Random random()` method to the `Simulation` and update this method to pull the simulation-wide random instead of creating a new one? This way the simulation remains deterministic but e don't need to plumb around metadata that shouldn't be otherwise meaningful.